cmd/register/cmd/internal/mock/store/store.go

```diff
@@ -0,0 +1,210 @@
+// Code generated by MockGen. DO NOT EDIT.
+// Source: internal/store/store.go
+
+// Package mock_store is a generated GoMock package.
+package mock_store
+
```

Comment:
The store mock could be moved to the project internal dir so it could be used also by other packages.

Reply:
I've introduced [gomock](https://github.com/golang/mock) and the store.go is generated using `gomock`, would like to hear from you on adopting it in the project

Reply:
@jskswamy no problem adopting it. Currently we focused on creating a lot integration tests to test real world scenarios so we didn't had the need to mock the store.